src/BL/WeatherForecast.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BL
+{
+    public class WeatherForecast
+    {
+        private static string apiKey = "c1c4d772f711221a4a1df5be101bb4a5";
+        private static string url1 = "https://api.openweathermap.org/data/2.5/weather?q=";
+        //private static string city = String.Empty;
+        private static string url2 = "&appid=";
+        private static string units = "&units=metric";
+        public static async Task<float> GetTemperature(string city)
+        {
+            string finalUrl = $"{url1}{city}{url2}{apiKey}{units}";
+            using (var client = new HttpClient())
+            {
+                HttpResponseMessage result = await client.GetAsync(finalUrl);
+
+                if (result.IsSuccessStatusCode)
+                {
+                    var json = result.Content.ReadAsStringAsync().Result;
+                    JObject obj = JsonConvert.DeserializeObject<JObject>(json);
+                    JObject mainObj = obj["main"] as JObject;
+                    float temperature = (float)mainObj["temp"];
+                    return temperature;
+                }
+                else
+                {
+                    return 0.0f;
```

Comment:
For city `Foo`, I get 0 temperature. You either need to throw exception or return null.

Reply:
Changed return 0.0f to return null; have to convert float to float?